### ISSUE TITLE
investigating the causes of unexpected integrity attestation failures

### ISSUE THREAD

**Issue by michaelficarra** [RupertBenWiser/Web-Environment-Integrity#12]
Up until this point, I've assumed the integrity decision was a single pass/fail bit. But in the case of failure, as we've discussed in previous CG calls, it is handy to have some way to investigate why an integrity failure happened for a transaction (or possibly many transactions) that we believe otherwise to have come from a trustable device. What could we do to be more informative about failures? Can the integrity decision contain some information about the reason for failure?

**Comment by RupertBenWiser** [comment]
Let me know if I've misunderstood the initial description, but this sounds to me like the debuggability aspect in the explainer. Is that accurate?

The assumption is that the attesters may not always get it right, and with fraud analysis, we need a way to report that back. I think that matters for both false negatives, and false positives.

**Comment by michaelficarra** [comment]
Yes, it is related to debuggability in the sense mentioned in the explainer. I'm not thinking about an attester "getting it wrong", but more "this device fails attestation and we don't know why". With only a pass/fail provided by the attester, there's not much you can do as a device or service administrator, especially if it only fails intermittently or under certain conditions. I think we need to find some middle ground between a single-bit pass/fail attestation and a full fingerprintable description of the operating conditions of the device.

**Comment by RupertBenWiser** [comment]
Ah gotcha, so this issue is about entropy of the attestation response - eg: Return "device rooted" vs "Low trust" and devices being able to self determine attestation failures? Yeah that will be a tricky problem to solve because reporting more information could also help adversarial parties.

The way I've been thinking about debuggability up to this point is ultimately for attesters to re-assess their own attestations. Let me know if I understand your concern correctly. Would also be good to hear your thoughts on if there are use cases you are concerned about where developers and users might need to debug their own devices.

**Comment by npdoty** [comment]
Debugging one's own device in the false positive case (this is a valid device, but it's failing attestation in some cases) seems well-suited to direct user-to-attester communication, like a page to report problems or a customer service ticket. The user is often willing to volunteer their data, and it's not specific to an origin (at least for models where the issuance protocol is separated from the origin).

Would that be useful to standardize? So that the user/device administrator knows where to go when they suspect a false positive?
